frame/alliance/src/lib.rs

```diff
@@ -483,12 +496,12 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
-	/// A set of members that are (potentially) being kicked out. They cannot retire until the
-	/// motion is settled.
+	/// A set of members who gave a retirement notice. They can retire after the end of retirement
+	/// period stored as a future block number.
 	#[pallet::storage]
-	#[pallet::getter(fn up_for_kicking)]
-	pub type UpForKicking<T: Config<I>, I: 'static = ()> =
-		StorageMap<_, Blake2_128Concat, T::AccountId, bool, ValueQuery>;
+	#[pallet::getter(fn retiring_members)]
+	pub type RetiringMembers<T: Config<I>, I: 'static = ()> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, T::BlockNumber, OptionQuery>;
```

Comment:
Mostly just curious, why not store this in the `MemberRole` variant itself (i.e. `Retiring(T::BlockNumber)`)? I'm not sure if we have FRAME guidance on this (@kianenigma or @shawntabrizi ?).

A storage item makes it easier for UIs to query all retiring members, but they could also periodically just check all the members and build that info from there.

Reply:
Right now there is one storage map to store all members, where the key is a member role and value is map of AccountIds (`Members`). 
And this additional map to store the BlockNumber for retirement period.